### PR TITLE
Use boost-cpp

### LINKED ISSUE #1: Splitting Python portion into a separate package
So, I recently hit a painful problem and am now seriously considering following Homebrew's path of breaking Boost into a non-Python package and having a Boost.Python package for just that part. The reason I am thinking about doing this is that it is causing me trouble when I build with SCons and require Boost. The net effect I fear is we have some package that becomes pinned in some artificial manner against Python 2.x when it has no Python dependency outside of Boost.

In particular, I am working on a build of MongoDB. See this PR ( https://github.com/conda-forge/staged-recipes/pull/392 ). I would be very curious to here others' thoughts about this proposal and if they have any other ideas about how to proceed.

### PR DESCRIPTION
**Pull request by isuruf**
Fixes #1

### CODE REVIEW

**Comment by jschueller** [comment]
should you not disable all other sub-libs ?

File: recipe/build.sh [hunk 1]
```diff
@@ -45,6 +45,7 @@ LINKFLAGS="${LINKFLAGS} -L${LIBRARY_PATH}"
     cxxflags="${CXXFLAGS}" \
     linkflags="${LINKFLAGS}" \
     --layout=system \
+    --with-python \
```
Comment (isuruf) [inline recipe/build.sh]: @jschueller, when this is given only python lib is built. You can see that in the logs.

Comment (jschueller) [inline recipe/build.sh]: oh ok sorry

**Comment by isuruf** [comment]
I'm getting a weird error in appveyor x64 builds. Any suggestions?

```
[00:02:43] Downloading http://sourceforge.net/projects/boost/files/boost/1.63.0/boost_1_63_0.tar.bz2
[00:02:54] Success
[00:03:32] Traceback (most recent call last):
[00:03:32]   File "C:\Miniconda-x64\Scripts\conda-build-script.py", line 5, in <module>
[00:03:32]     sys.exit(conda_build.cli.main_build.main())
[00:03:32]   File "C:\Miniconda-x64\lib\site-packages\conda_build\cli\main_build.py", line 312, in main
[00:03:32]     execute(sys.argv[1:])
[00:03:32]   File "C:\Miniconda-x64\lib\site-packages\conda_build\cli\main_build.py", line 304, in execute
[00:03:32]     already_built=None, config=config, noverify=args.no_verify)
[00:03:32]   File "C:\Miniconda-x64\lib\site-packages\conda_build\api.py", line 86, in build
[00:03:32]     need_source_download=need_source_download, config=config)
[00:03:32]   File "C:\Miniconda-x64\lib\site-packages\conda_build\build.py", line 1427, in build_tree
[00:03:32]     config=config)
[00:03:32]   File "C:\Miniconda-x64\lib\site-packages\conda_build\build.py", line 948, in build
[00:03:32]     config=config)
[00:03:32]   File "C:\Miniconda-x64\lib\site-packages\conda_build\render.py", line 87, in parse_or_try_download
[00:03:32]     source.provide(metadata, config=config)
[00:03:32]   File "C:\Miniconda-x64\lib\site-packages\conda_build\source.py", line 498, in provide
[00:03:32]     unpack(metadata, config=config)
[00:03:32]   File "C:\Miniconda-x64\lib\site-packages\conda_build\source.py", line 88, in unpack
[00:03:32]     tar_xf(src_path, config.work_dir)
[00:03:32]   File "C:\Miniconda-x64\lib\site-packages\conda_build\utils.py", line 313, in tar_xf
[00:03:32]     t.extractall(path=dir_path)
[00:03:32]   File "C:\Miniconda-x64\lib\tarfile.py", line 2079, in extractall
[00:03:32]     self.extract(tarinfo, path)
[00:03:32]   File "C:\Miniconda-x64\lib\tarfile.py", line 2116, in extract
[00:03:32]     self._extract_member(tarinfo, os.path.join(path, tarinfo.name))
[00:03:32]   File "C:\Miniconda-x64\lib\tarfile.py", line 2192, in _extract_member
[00:03:32]     self.makefile(tarinfo, targetpath)
[00:03:32]   File "C:\Miniconda-x64\lib\tarfile.py", line 2232, in makefile
[00:03:32]     with bltn_open(targetpath, "wb") as target:
[00:03:32] IOError: [Errno 2] No such file or directory: 'C:\\Miniconda-x64\\conda-bld\\recipe_1483772488890\\work\\boost_1_63_0\\libs\\geometry\\doc\\html\\geometry\\reference\\spatial_indexes\\boost__geometry__index__rtree\\rtree_parameters_type_const____indexable_getter_const____value_equal_const____allocator_type_const___.html'
[00:03:33] Command exited with code 1
```

**Comment by jschueller** [comment]
file path limit ?
this is pretty long: ```boost_1_63_0/libs/geometry/doc/html/geometry/spatial_indexes/creation_and_modification.html```
maybe it fails on x64 because the location crossed the limit since there's a "-x64" suffix in CONDA_INSTALL_LOCN=C:\\Miniconda-x64

**Comment by isuruf** [comment]
Probably. without `-x64` it is 256 chars long and with `-x64` it is 260 chars long.
What I don't get is why it didn't fail before.

**Comment by jschueller** [comment]
it was conda-build 1.x ?

**Comment by isuruf** [comment]
Yes, that seems to be the case. @msarahan, any suggestions?

**Comment by jschueller** [comment]
@isuruf that's definitively it: the build dir changed from conda-bld/work from conda-bld/recipe_[unixtime] which is much longer, got the same problem for another recipe

**Comment by isuruf** [comment]
Thanks @jschueller for the info. Can you open an issue in conda-build ?

**Comment by jakirkham** [comment]
We just updated to `conda-build` 2 the other day. Undoubtedly there may be some issues we uncover. However, unlike `conda-build` 1 we will get better support.

**Comment by isuruf** [comment]
It works on my fork, so I created conda-forge/conda-smithy/pull/419

**Comment by isuruf** [comment]
@jakirkham, Look at the last build.
```
C:\bld\recipe_1484368660609       # conda-build 2.x with this PR
C:\Miniconda3-x64\conda-bld       # conda-build 1.x
```

**Comment by jakirkham** [comment]
Nice! Looks good. Let's add that to `conda-forge-build-setup`. Could you please do a PR?

**Comment by isuruf** [comment]
Deleted the changes for build directory as it is now in conda-forge-build-setup.
Tests pass now, except for one test in appveyor which fails due to a corrupt download.

**Comment by jakirkham** [comment]
Appears to have passed on `master`. Thanks for doing this, @isuruf.

**Comment by jakirkham** [comment]
Though it does seem we are missing the headers for Boost.Python in Linux and OS X packages. Windows packages seem to include all the headers for all of Boost. 😕

**Pull request merged**